File: Codeforces Round #828 (Div. 3)/b.cpp
#include <iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n, q;
        cin >> n >> q;
        long long a[n];
        long long sum = 0;
        int e = 0;
        for (int i = 0; i < n; i++)
        {
            cin >> a[i];
            sum += a[i];
            if (a[i] % 2 == 0)
            {
                e++;
            }
        }
        int o = n - e;
        for (int i = 0; i < q; i++)
        {
            int d, v;
            cin >> d >> v;
            if (d == 0)
            {
                sum += (v * e);
                if (v % 2 != 0)
                {
                    int temp = e;
                    e -= temp;
                    o += temp;
                }
            }
            else
            {
                sum += (v * o);
                if (v % 2 != 0)
                {
                    int temp = o;
                    o -= temp;
                    e += temp;
                }
            }
            cout << sum << endl;
        }
    }
    return 0;
}

File: Codeforces Round #828 (Div. 3)/b.py
t = int(input())
for _ in range(t):
    n, q_no = map(int, input().split())
    a = list(map(int, input().split()))
    sum_a = sum(a)
    e = 0  # evens
    for i in range(len(a)):
        if a[i] % 2 == 0:
            e += 1
    o = len(a) - e  # odds
    for _ in range(q_no):
        d, v = map(int, input().split())
        if d == 0:
            t_sum = sum_a + (v * e)
            sum_a = t_sum
            if v % 2 != 0:
                temp = e
                e -= temp
                o += temp
        else:
            t_sum = sum_a + (v * o)
            sum_a = t_sum
            if v % 2 != 0:
                temp = o
                o -= temp
                e += temp
        print(sum_a)

